demos/embeddings/barbell-graphwave.ipynb

```diff
@@ -0,0 +1,167 @@
+      "text/plain": [
+       "<Figure size 432x288 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "m1 = 10\n",
+    "m2 = 11\n",
+    "gnx = nx.barbell_graph(m1=m1, m2=m2)\n",
+    "\n",
+    "classes = [0,] * len(gnx.nodes)\n",
+    "# number of nodes with a non-zero class (the path, plus the nodes it connects to on either end)\n",
+    "nonzero = m2 + 2\n",
+    "# count up to the halfway point (rounded up)\n",
+    "first = range(1, (nonzero + 1) // 2 + 1)\n",
+    "# and down for the rest\n",
+    "second = reversed(range(1, nonzero - len(first) + 1))\n",
+    "classes[m1 - 1 : (m1 + m2) + 1] = list(first) + list(second)\n",
+    "\n",
+    "nx.draw(gnx, node_color=classes, cmap=\"jet\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### GraphWave embedding calculation\n",
+    "\n",
+    "Now, we're ready to calculate the GraphWave embeddings. We need to specify some information about how to sample the characteristic function: \n",
+    "\n",
+    "- an iterable of wavelet `scales` to use. This is a graph dependent hyperparameter. Experiment with different values.\n",
+    "- the number `num_eigenvecs` of eigenvectors to use. A larger number of eigenvectors will yield better results but is more computationally expensive. `num_eigenvecs=None` will automatically estimate and use a number of eigenvectors that is a reasonable approximation to the full set of eigenvectors.\n",
+    "- the `sample_points` at which to sample the characteristic function. This should be of the form `sample_points=np.linspace(0, max_val, number_of_samples)`. The best value depends on the graph."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "G = StellarGraph(gnx)\n",
+    "sample_points = np.linspace(0, 100, 50).astype(np.float32)\n",
+    "\n",
+    "generator = GraphWaveGenerator(G, scales=[5, 10], num_eigenvecs=None,)\n",
+    "\n",
+    "embeddings_dataset = generator.flow(\n",
+    "    node_ids=G.nodes(), sample_points=sample_points, batch_size=1, repeat=False\n",
+    ")\n",
+    "\n",
+    "embeddings = [x.numpy() for x in embeddings_dataset]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Visualation\n",
+    "\n",
+    "The nodes are coloured by their structural role, e.g. in the connected component, 1st node in the chain, 2nd node in the chain etc. We can see that all nodes of the same colour completely overlap in this visualation, indicating that structurally equivalent nodes are very close in the embedding space."
```

Comment:
```suggestion
    "The nodes are coloured by their structural role, e.g. in the fully connected sections, 1st node in the chain, 2nd node in the chain etc. We can see that all nodes of the same colour completely overlap in this visualation, indicating that structurally equivalent nodes are very close in the embedding space."
```

because the whole graph is a single connected component.